src/cluster.c

```diff
@@ -1083,24 +1083,17 @@ ShardGroup **ShardGroupsParse(RedisModuleCtx *ctx,
  * ShardGroup.
  */
 
-void ShardingInfoInit(RedisRaftCtx *rr)
+void ShardingInfoInit(RedisModuleCtx *ctx, ShardingInfo **si)
```

Comment:
so I'm not quite sure why the prototype changed, it still only taking things from the RedisRaftCtx

Reply:
ok, now I think understand better (needs a better description to go over all that changes here).  I think is about simply making it separation of  concerns.  while, yes, it could just pass a single rr pointer, passing the RedisModuleCtx and a double pointer to the pointer we would alloc and handle is a "cleaner" api.